### PR TITLE
Improved canonical path for vision in Google.

### PR DESCRIPTION
**Pull request by ivan-nginx**
## PR Checklist
**Please check if your PR fulfills the following requirements:**

- [x] The commit message follows [our guidelines](https://github.com/theme-next/hexo-theme-next/blob/master/.github/CONTRIBUTING.md).
- [x] Tests for the changes was maked (for bug fixes / features).
   - [x] Muse | Mist have been tested.
   - [x] Pisces | Gemini have been tested.
- [x] Docs in [NexT website](https://theme-next.org/docs/) have been added / updated (for new features).

## PR Type
**What kind of change does this PR introduce?**

- [ ] Bugfix.
- [ ] Feature.
- [ ] Code style update (formatting, local variables).
- [ ] Refactoring (no functional changes, no api changes).
- [ ] Build related changes.
- [ ] CI related changes.
- [ ] Documentation content changes.
- [x] Other... Please describe: **improvement**

## What is the current behavior?
Pages must placed in subdirectories with `index.html` names, e.g. `source/docs/troubleshooting/index.md` for pretty path with `canonical` option enabled:
image

But if we want to use pages in root directories and named as is with `*.html` suffix, `source/docs/advanced-settings.md`, canonical path will contain `*.html` suffix:
image

## What is the new behavior?
From now all links will be not only without `index.html`, but without `.html` too for pretty path with `canonical` option enabled. **And not need to set `permalink` in each page for that**.

Pages now can placed in root directories and named as is with `*.html` suffix, e.g. `source/docs/advanced-settings.md`:
image

- Link to demo site with this changes: https://theme-next.org/docs/advanced-settings
  (with https://github.com/theme-next/theme-next.org/commit/de8fffcfb14bdab4945fe711ee6bbb2a0866ef78 and https://github.com/theme-next/theme-next.org/commit/ac9141c1eba304d9c727226b576a6822680c617d changes)
- Hexo have same behavior: https://hexo.io/docs/setup

### How to use?
In local servers `*.html` suffix will be exists. For remove they in Production, need to create redirects (for Nginx) or use advanced settings in deployment services (like «Pretty URLs» in Netlify):
image

## Does this PR introduce a breaking change?
- [ ] Yes.
- [x] No.

### CODE REVIEW

**Comment by 1v9** [comment]
For links contain html like https://1v9.im/post/theme-next-added-today's-poetry.html
canonical will be: `<link rel="canonical" href="https://1v9.im/post/theme-next-added-today's-poetry">`

**Comment by ivan-nginx** [comment]
> For links contain html like https://1v9.im/post/theme-next-added-today's-poetry.html
> canonical will be: `<link rel="canonical" href="https://1v9.im/post/theme-next-added-today's-poetry">`

U just updated readme, reread it.

File: _config.yml [hunk 1]
```diff
@@ -185,7 +185,7 @@ scheme: Muse
 
 # ---------------------------------------------------------------
 # Sidebar Settings
-# See: https://theme-next.org/docs/theme-settings/sidebar/
+# See: https://theme-next.org/docs/theme-settings/sidebar
```
Comment (stevenjoezhang) [inline _config.yml]: Wait a minute... this url is for users, not search engine. If users copy & past `https://theme-next.org/docs/theme-settings/sidebar` in their browser, the browser will request `https://theme-next.org/docs/theme-settings/sidebar`, and then receive a 301 rewrite to `https://theme-next.org/docs/theme-settings/sidebar/`, and the browser will request again. Which means: delete the `/` at the end of the url will cause an extra http request.

Comment (ivan-nginx) [reply _config.yml]: What are u talking about?
https://check-host.net/check-http?host=https://theme-next.org/docs/advanced-settings

**Pull request merged**